php/php.editor/src/org/netbeans/modules/php/editor/verification/ReturnTypeHintError.java

```diff
@@ -152,17 +159,40 @@ private void checkVoidAndNeverReturnStatements(Set<ReturnStatement> statements,
         "ReturnTypeHintErrorInvalidVoidDesc=\"{0}\" cannot be used with \"?\""
     })
     private void checkInvalidVoidAndNeverReturnType(Expression returnType, String name, List<Hint> hints) {
-        if (Type.VOID.equals(name)
+        if (isVoidType(name)
                 || isNeverType(name)) {
             addHint(returnType, Bundle.ReturnTypeHintErrorInvalidVoidDesc(name), hints);
         }
     }
 
+    @NbBundle.Messages({
+        "ReturnStatementWithoutValueHintErrorDesc=Return statement must be filled with the value"
+    })
+    private void checkReturnStatementsWithoutValue(Set<ReturnStatement> statements, String name, List<Hint> hints) {
+        if (!isVoidType(name) && !isNeverType(name)) {
+            // check empty return statement
+            for (ReturnStatement statement: statements) {
+                if (CancelSupport.getDefault().isCancelled()) {
+                    return;
+                }
+                Expression expression = statement.getExpression();
+                if (expression == null) {
+                    addHint(statement, Bundle.ReturnStatementWithoutValueHintErrorDesc(), hints);
+                }
+            }
+        }
+    }
+
     private boolean isNeverType(String name) {
         return getPhpVersion(fileObject).hasNeverType()
                 && Type.NEVER.equals(name);
     }
 
+   private boolean isVoidType(String name) {
+        return getPhpVersion(fileObject).hasVoidReturnType()
+                && Type.VOID.equals(name);
```

Comment:
I am not sure, but I would expect that if we swap the conditions, the performance should be better (comparing strings should be faster than working with file objects).

Reply:
(It applies also for `isNeverType` method.)

Reply:
Done.